app/models/lockstep/service_fabric_payment.rb

```diff
@@ -3,5 +3,34 @@ class Lockstep::ServiceFabricPayment < Lockstep::ApiRecord
   self.id_ref = "end_to_end_identifier"
   self.query_path=""
   load_schema(Schema::SfPaymentRequest)
+  
+  # Returning alwas false becase if id_ref is present in
+  # request body it returns true but in this model id_ref
+  # is required field in request body
+  # TODO: Make it generic 
+  def persisted?
+    false
+  end
+
+  # Preventing deletion of id_ref from puts_attrs because
+  # id_ref column is required to send in body in case of
+  # create ServiceFabricPayment
+  # TODO: Make it generic
+  def attributes_for_saving
```

Comment:
this is the same thing in the apirecord.rb class right , how is this method modified and can you let us know the scenario for this override as well

Reply:
Here I removed `put_attrs.delete(id_ref)` to not delete `id_ref` key from request body